packages/webapp/src/delegates/api/eden-contract.ts

```diff
@@ -1,31 +1,92 @@
-import { getElectionState } from "elections/api";
+import { EdenMember } from "members";
+import { queryClient } from "pages/_app";
+import {
+    queryElectionState,
+    queryHeadDelegate,
+    queryMemberByAccountName,
+} from "_app";
 
-const getDelegates = () => {
-    return {}; // TODO
+const queryElectionStateHelper = async () => {
+    try {
+        return await queryClient.fetchQuery(
+            queryElectionState.queryKey,
+            queryElectionState.queryFn
+        );
+    } catch (e) {
+        console.error("Error in getMyDelegation:");
+        console.error(e);
+        return Promise.resolve(undefined);
```

Comment:
I think just returning undefined will resolve the async promise here.

Nonetheless, I'm somewhat uncomfortable with us swallowing all of these errors. React Query expects the async functions it wraps in queries to _throw_. In fact, they even give guidance on how to make async functions that don't normally throw (like `fetch`) to do so: https://react-query.tanstack.com/guides/query-functions#handling-and-throwing-error

I recommend simply removing the `try...catch` here. We're already dealing with the lack of data (`undefined`) in the UI in the event one of these throws. And that just gives us the added option of intercepting and handling errors there in the UI too.

Reply:
👍